lib/rubocop/cop/performance/map_compact.rb

```diff
@@ -0,0 +1,65 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Performance
+      # In Ruby 2.7, `Enumerable#filter_map` has been added.
+      #
+      # This cop identifies places where `map { ... }.compact` can be replaced by `filter_map`.
+      # It is marked as unsafe auto-correction by default because `map { ... }.compact`
+      # that is not compatible with `filter_map`.
+      #
+      # [source,ruby]
+      # ----
+      # [true, false, nil].compact              #=> [true, false]
+      # [true, false, nil].filter_map(&:itself) #=> [true]
+      # ----
+      #
+      # @example
+      #   # bad
+      #   ary.map(&:foo).compact
+      #   ary.collect(&:foo).compact
+      #
+      #   # good
+      #   ary.filter_map(&:foo)
+      #   ary.map(&:foo).compact!
+      #
+      class MapCompact < Base
+        include RangeHelp
+        extend AutoCorrector
+        extend TargetRubyVersion
+
+        MSG = 'Use `filter_map` instead.'
+        RESTRICT_ON_SEND = %i[compact].freeze
+
+        minimum_target_ruby_version 2.7
+
+        def_node_matcher :map_compact?, <<~PATTERN
```

Comment:
Since this returns a node, and not a boolean, shall we cut the `?` suffix? 🙂

Reply:
Oh, this is my mistake. Thank you for your review :-)